server/apps/users/views.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from django.shortcuts import render
+from django.conf import settings
+from django.shortcuts import redirect, render
+from django.contrib import messages
+from django.forms.models import model_to_dict
+
+from .models import UserProfile
+from .forms import UserProfileForm
+from .helpers import (
+    user_profile_exists,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def user_profile(request):
+    """
+    User profile page.
+    """
+    if request.user.is_authenticated:
+        if request.method == "POST":
+            form = UserProfileForm(request.POST)
+
+            if form.is_valid():
+                # Update the status to indicate the user has submitted at least once
+                form.cleaned_data["profile_submitted"] = True
+                UserProfile.objects.update_or_create(user=request.user, defaults=form.cleaned_data)
+
+            return redirect("users:profile")
+        else:
+            first_visit = not user_profile_exists(request.user)
+            profile, created = UserProfile.objects.get_or_create(user=request.user)
+            data = model_to_dict(profile)
+            form = UserProfileForm(data)
+            oh_member = request.user.openhumansmember
+            return render(
```

Comment:
I think the `title` context is missing to make sure the template renders the part after _AutSPACEs –_ as well

Reply:
Thank you! I've given it a title now.